diff --git a/node_modules/@astrojs/starlight/utils/navigation.ts b/node_modules/@astrojs/starlight/utils/navigation.ts
index 526d9f2..de8c64a 100644
--- a/node_modules/@astrojs/starlight/utils/navigation.ts
+++ b/node_modules/@astrojs/starlight/utils/navigation.ts
@@ -360,8 +360,12 @@ export function getSidebarFromConfig(
 	locale: string | undefined
 ): SidebarEntry[] {
 	const routes = getLocaleRoutes(locale);
+	const currentSection = pathname.split("/")[1];
 	if (sidebarConfig) {
-		return sidebarConfig.map((group) => configItemToEntry(group, pathname, locale, routes));
+		return sidebarConfig.flatMap((group) => {
+			if (group.label !== currentSection) return [];
+			return configItemToEntry(group, pathname, locale, routes)
+		});
 	} else {
 		const tree = treeify(routes, locale || '');
 		return sidebarFromDir(tree, pathname, locale, false);
